src/renderers/dom/shared/inputValueTracking.js

```diff
@@ -113,40 +113,46 @@ var inputValueTracking = {
     return getTracker(ReactDOMComponentTree.getInstanceFromNode(node));
   },
 
-  trackNode: function(node: ElementWithWrapperState) {
-    if (node._wrapperState.valueTracker) {
+  trackNode(node: ElementWithWrapperState) {
+    if (getTracker(node)) {
       return;
     }
     node._wrapperState.valueTracker = trackValueOnNode(node, node);
   },
 
-  track: function(inst: InstanceWithWrapperState) {
+  track(inst: InstanceWithWrapperState) {
     if (getTracker(inst)) {
       return;
     }
     var node = ReactDOMComponentTree.getNodeFromInstance(inst);
-    attachTracker(inst, trackValueOnNode(node, inst));
+    inst._wrapperState.valueTracker = trackValueOnNode(node, inst);
   },
 
-  updateValueIfChanged(inst: InstanceWithWrapperState | Fiber) {
-    if (!inst) {
+  updateValueIfChanged(subject: SubjectWithWrapperState | Fiber) {
+    if (!subject) {
       return false;
     }
-    var tracker = getTracker(inst);
+    var tracker = getTracker(subject);
 
     if (!tracker) {
-      if (typeof (inst: any).tag === 'number') {
-        inputValueTracking.trackNode((inst: any).stateNode);
+      if (typeof (subject: any).tag === 'number') {
+        inputValueTracking.trackNode((subject: any).stateNode);
       } else {
-        inputValueTracking.track((inst: any));
+        inputValueTracking.track((subject: any));
```

Comment:
This branching looks odd to me. It seems like it tries to branch on Fiber and Stack code. However now `subject` could be a DOM node itself. In this case it will go to `track()` rather than `trackNode()` which is why it breaks.

I think this shows why it might be better to reduce the polymorphism here and maybe change this to only accept DOM nodes. But I'd also like to understand why tests (including fixtures) didn't catch this. It is not quite obvious to me.